test/src/api/commands/element/testIsVisible.js

```diff
@@ -57,4 +65,74 @@ describe('isVisible', function () {
 
     this.client.start(done);
   });
+
+  it('client.isVisible() [visible] -- appium', function (done) {
+    MockServer
+      .addMock({
+        url: '/wd/hub/session/1352110219202/elements',
+        postdata: {
+          using: 'id',
+          value: 'com.app:id/weblogin'
+        },
+        method: 'POST',
+        response: JSON.stringify({
+          status: 0,
+          state: 'success',
+          value: [{'element-6066-11e4-a52e-4f735466cecf': '999'}]
+        })
+      })
+      .addMock({
+        url: '/wd/hub/session/1352110219202/element/999/displayed',
+        method: 'GET',
+        response: JSON.stringify({
+          value: true
+        })
+      });
+
+    // Make appium client
+    this.client.api.options.selenium.use_appium = true;
+
+    this.client.api.isVisible('id', 'com.app:id/weblogin', function callback(result) {
+      assert.strictEqual(result.value, true);
+    }).isVisible({selector: 'com.app:id/weblogin', locateStrategy: 'id'}, function callback(result) {
+      assert.strictEqual(result.value, true);
+    });
+
+    this.client.start(done);
+  });
+
+  it('client.isVisible() [not visible] -- appium', function (done) {
+    MockServer
+      .addMock({
+        url: '/wd/hub/session/1352110219202/elements',
+        postdata: {
+          using: 'id',
+          value: 'com.app:id/weblogin'
+        },
+        method: 'POST',
+        response: JSON.stringify({
+          status: 0,
+          state: 'success',
+          value: [{'element-6066-11e4-a52e-4f735466cecf': '999'}]
+        })
+      })
```

Comment:
why do we need separate mocks for elements? We can use the existing mock added globally

Reply:
The mocks that we have available globally consider the `locateStrategy` to be 'css selector' and hence, do not work with Appium tests. And I had earlier declared this mock in the `mocks-w3c.yaml` only, but then in all other tests I anyway had to write separate mocks so I also moved this mock to the test file only.